machines/Makefile

```diff
@@ -1,50 +1,44 @@
-
-ifndef BSG_MANYCORE_DIR
-  export BSG_MANYCORE_DIR := $(call find-dir-with,.BSG_MANYCORE_ROOT)
-endif
+.DEFAULT_GOAL = all
 
+export BSG_MANYCORE_DIR := $(shell git rev-parse --show-toplevel)
 export BASEJUMP_STL_DIR := $(BSG_MANYCORE_DIR)/../basejump_stl
 
 # Sets "VCS" variable
 include $(BSG_MANYCORE_DIR)/../bsg_cadenv/cadenv.mk
 
-MACHINES = 9x15 4x4
-
-SIMV = $(addsuffix /simv, $(MACHINES))
-SIMV_DEBUG = $(addsuffix /simv-debug, $(MACHINES))
-SIMV_DAIDIR = $(addsuffix .daidir, $(SIMV) $(SIMV_DEBUG))
+# Include source lists
+include Makefile.include
 
-VCS_INCDIR := +incdir+$(BASEJUMP_STL_DIR)/bsg_misc
-VCS_INCDIR += +incdir+$(BASEJUMP_STL_DIR)/bsg_cache
-VCS_INCDIR += +incdir+$(BASEJUMP_STL_DIR)/bsg_noc
-VCS_INCDIR += +incdir+$(BSG_MANYCORE_DIR)/v
-VCS_INCDIR += +incdir+$(BSG_MANYCORE_DIR)/v/vanilla_bean
+MACHINES = 9x15 4x4
 
 .PHONY: all clean
 
-all: $(SIMV)
+all: $(foreach machine, $(MACHINES), $(machine)/stamp)
 
-%/simv:
+%/stamp: $(VCS_SOURCES) $(VCS_INCLUDES)
 	$(eval include $*/Makefile.machine.include)
-	$(eval VCS_DEFINE := +define+BSG_MACHINE_GLOBAL_X=${BSG_MACHINE_GLOBAL_X})
-	$(eval VCS_DEFINE += +define+BSG_MACHINE_GLOBAL_Y=${BSG_MACHINE_GLOBAL_Y})
-	$(eval VCS_DEFINE += +define+BSG_MACHINE_VCACHE_SET=${BSG_MACHINE_VCACHE_SET})
-	$(eval VCS_DEFINE += +define+BSG_MACHINE_VCACHE_WAY=${BSG_MACHINE_VCACHE_WAY})
-	$(eval VCS_DEFINE += +define+BSG_MACHINE_VCACHE_BLOCK_SIZE_WORDS=${BSG_MACHINE_VCACHE_BLOCK_SIZE_WORDS})
-	$(eval VCS_DEFINE += +define+BSG_MACHINE_DRAM_SIZE_WORDS=${BSG_MACHINE_DRAM_SIZE_WORDS})
-	$(eval VCS_DEFINE += +define+BSG_MACHINE_DRAM_INCLUDED=${BSG_MACHINE_DRAM_INCLUDED})
-	$(eval VCS_DEFINE += +define+BSG_MACHINE_MAX_EPA_WIDTH=${BSG_MACHINE_MAX_EPA_WIDTH})
-	$(eval VCS_DEFINE += +define+BSG_MACHINE_MEM_CFG=${BSG_MACHINE_MEM_CFG})
+	$(eval VCS_MACRO_DEFS := +define+BSG_MACHINE_GLOBAL_X=${BSG_MACHINE_GLOBAL_X})
+	$(eval VCS_MACRO_DEFS += +define+BSG_MACHINE_GLOBAL_Y=${BSG_MACHINE_GLOBAL_Y})
+	$(eval VCS_MACRO_DEFS += +define+BSG_MACHINE_VCACHE_SET=${BSG_MACHINE_VCACHE_SET})
+	$(eval VCS_MACRO_DEFS += +define+BSG_MACHINE_VCACHE_WAY=${BSG_MACHINE_VCACHE_WAY})
+	$(eval VCS_MACRO_DEFS += +define+BSG_MACHINE_VCACHE_BLOCK_SIZE_WORDS=${BSG_MACHINE_VCACHE_BLOCK_SIZE_WORDS})
+	$(eval VCS_MACRO_DEFS += +define+BSG_MACHINE_DRAM_SIZE_WORDS=${BSG_MACHINE_DRAM_SIZE_WORDS})
+	$(eval VCS_MACRO_DEFS += +define+BSG_MACHINE_DRAM_INCLUDED=${BSG_MACHINE_DRAM_INCLUDED})
+	$(eval VCS_MACRO_DEFS += +define+BSG_MACHINE_MAX_EPA_WIDTH=${BSG_MACHINE_MAX_EPA_WIDTH})
+	$(eval VCS_MACRO_DEFS += +define+BSG_MACHINE_MEM_CFG=${BSG_MACHINE_MEM_CFG})
 	$(VCS) +v2k -sverilog -full64 -o $*/simv \
 		+lint=all,noSVA-UA,noSVA-NSVU,noVCDE,noNS \
 		-timescale=1ps/1ps \
-		-f tb.include $(VCS_INCDIR) $(VCS_DEFINE) | tee -i $*/build.log
+		$(VCS_INCDIRS) $(VCS_DEFINES) $(VCS_SOURCES) $(VCS_MACRO_DEFS) | tee -i $*/build.log
```

Comment:
this should be called VCS_DEFINE

Reply:
Why?

Reply:
@tommydcjung I had used `VCS_DEFINES` for listing `bsg_defines.v` kind of files. So, had to change this to macro_defs.

Reply:
in that case, would it be more appropriate to call it VERILOG_DEFINES? instead of VCS_DEFINES? Because bsg_defines.v really has nothing to do with VCS.

Reply:
https://github.com/bespoke-silicon-group/bsg_manycore/blob/1f653f7532fbdd19e054c0fc96c942dc97f2d7ac/machines/Makefile.include#L10

Please refer to Dustin's comments about standardizing Make variable names.

Reply:
okay, is this standard documented somewhere? or does this rule change whenever someone makes a change?

Reply:
@tommydcjung the idea is to make this easy to port, when adding support for other tools. This is a step towards that needing little effort.